bin/lint-migrations-file/src/lint_migrations_file.clj

```diff
@@ -35,75 +35,63 @@
-    (when (map? x)
-      (cond
-       ;; a createTable or addColumn change; see if it adds a target-type col
-       (or (:createTable x) (:addColumn x))
-       (let [op     (cond (:createTable x) :createTable (:addColumn x) :addColumn)
-             cols   (filter (fn [col-def]
-                              (match-target-types? (get-in col-def [:column :type] "")))
-                            (get-in x [op :columns]))]
-         (doseq [col cols]
-           (swap! found-cols conj col)))
-
-
-       ;; a modifyDataType change; see if it changes a column to target-type
-       (and (:modifyDataType x)
-            (match-target-types? (get-in x [:modifyDataType :newDataType] "")))
-       (swap! found-cols conj x))
-      x)))
+                              (contains? types (str/lower-case ttype)))]
+    (cond
+     ;; a createTable or addColumn change; see if it adds a target-type col
+     (or (:createTable change) (:addColumn change))
+     (let [op (cond (:createTable change) :createTable (:addColumn change) :addColumn)]
+       (some (fn [col-def]
+               (match-target-types? (get-in col-def [:column :type] "")))
+             (get-in change [op :columns])))
+
+     ;; a modifyDataType change; see if it change a column to target-type
+     (and (:modifyDataType change)
+          (match-target-types? (get-in change [:modifyDataType :newDataType] "")))
+     true)))
+
+(defn- check-change-set-use-types?
+  "Return true if `change-set` doesn't contain usage of any type in `types`."
+  [target-types change-set]
+  (some #(check-change-use-types? target-types %) (get-in change-set [:changeSet :changes])))
+
+(defn- check-change-log-use-types?
+  ([target-types change-log]
+   (check-change-log-use-types? target-types change-log (constantly true)))
+  ([target-types change-log id-filter-fn]
+   {:pre [(set? target-types)]}
+   (->> change-log
+        (filter (fn [change-set]
+                  (let [id (get-in change-set [:changeSet :id])]
+                    (and (string? id)
+                         (id-filter-fn id)))))
+        (some #(check-change-set-use-types? target-types %)))))
 
 (defn no-bare-blob-or-text-types?
   "Ensures that no \"text\" or \"blob\" type columns are added in changesets with id later than 320 (i.e. version
   0.42.0).  From that point on, \"${text.type}\" should be used instead, so that MySQL can handle it correctly (by using
   `LONGTEXT`).  And similarly, from an earlier point, \"${blob.type}\" should be used instead of \"blob\"."
   [change-log]
-  (let [problem-cols (atom [])
-        walk-fn      (partial assert-no-types-in-change-set #{"blob" "text"} problem-cols)]
-    (doseq [{{id :id} :changeSet :as change-set} change-log
-            :when                                (and id
-                                                      (string? id)
-                                                      (str/starts-with? id "v"))]
```

Comment:
Can you explain why you removed the `(str/starts-with? id "v")` check?

Reply:
all the migrations in `001_update_migrations.yaml` now start with `v`; it was needed when we still had the integer ID in the same migration file.